{{cookiecutter.project_slug}}/.github/workflows/docs.yml

```diff
@@ -0,0 +1,42 @@
+name: Documentation
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+      - name: Cache tox
+        uses: actions/cache@v4
+        with:
+          path: .tox
+          key: tox-${{ '{{' }} hashFiles('pyproject.toml') {{ '}}' }}
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+          cache: "pip"
+          cache-dependency-path: "pyproject.toml"
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Build HTML documentation with tox
+        run: tox -e docs
+    # Uncomment lines below to set-up automatic deployment of documentation to a
+    # GitHub Pages website on pushing to main - you will need configure the repository
+    # to deploy from a branch gh-pages (https://tinyurl.com/gh-pages-from-branch),
+    # which will be created the first time this workflow step is run
+    #   - name: Publish documentation on GitHub pages
+    #     if: success() && github.event_name != 'pull_request'
+    #     uses: peaceiris/actions-gh-pages@v3
+    #     with:
+    #       github_token: ${{ '{{' }}  secrets.GITHUB_TOKEN {{ '}}' }}
+    #       publish_dir: site
+    #       publish_branch: gh-pages
+    #       user_name: "github-actions[bot]"
+    #       user_email: "github-actions[bot]@users.noreply.github.com"
```

Comment:
What's your plan with this section?

Reply:
By default to just leave this as commented in the template and add something to top-level README about uncommenting if you want to enable automatic deployment of documentation to a GitHub Pages site. This seems similar to how we already handle the commented out PyPI / conda badges in README. We could instead add an extra option to `cookiecutter.json` to ask whether to include, but we'd probably also need to add a post processing hook or some other approach for printing a message similar to the comment about setting up GitHub pages configuration for the repository

Reply:
I'm pretty against the idea of permanently commented out code. Makes me think of classic non-versioned-controlled academic code.

Reply:
I've addded an additional option to `cookiecutter.json` `deploy_docs_to_github_pages` (with prompt _Enable automatically deploying HTML documentation to GitHub Pages website on pushes to main_) that allows selectively including the additional job step as an opt-in (defaults to disabled). Still requires user to set up GitHub Pages to deploy from `gh-pages` branch - for now I've just left an explanatory comment in workflow to this effect.

Reply:
Relates to #205